sklearn/metrics/tests/test_classification.py

```diff
@@ -248,6 +248,15 @@ def test_matthews_corrcoef_nan():
     assert_equal(matthews_corrcoef([0, 0], [0, 1]), 0.0)
 
 
+def test_matthews_corrcoef_result():
+    rng = np.random.RandomState(0)
+    y_true = rng.randint(0, 2, size=20)
+    y_pred = rng.randint(0, 2, size=20)
+
+    assert_almost_equal(matthews_corrcoef(y_true, y_pred),
+                        np.corrcoef(y_true, y_pred)[0, 1], 10)
```

Comment:
the numpy tests on corrcoef were mostly about missing values (which we don't allow), 2d (which is not applicable for the binary classification task for which this metric is intended), and other minor usecases which were not applicable here...

But I added a few more tests which check 
* If mcc for same arrays is 1
* mcc for opp arrays is -1
* mcc for 2 uncorrelated arrays  is 0
* mcc for zero vector raises `RuntimeWarning`, but still outputs 0
* If sample weight has any effect by checking if the mcc for the uncorrelated arrays, with only half of the values weighted, is not 0

I hope the  above tests are sufficient? (you can selectively review the last commit for the tests, if you want)